MapboxNavigationUI/RouteVoiceController.swift

```diff
@@ -242,16 +242,23 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
         return text
     }
     
-    func localizeRoadDescription(_ step: RouteStep) -> String {
+    func localizeRoadDescription(_ step: RouteStep, markUpWithSSML: Bool) -> String {
         var road = ""
+        let escapeIfNecessary = {(distance: String) -> String in
+            return markUpWithSSML ? distance.addingXMLEscapes : distance
+        }
         if let name = step.names?.first {
             if let code = step.codes?.first {
-                road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name, code)
+                if markUpWithSSML {
+                    road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "<say-as interpret-as=\"address\">%@</say-as> (<say-as interpret-as=\"address\">%@</say-as>)", comment: "Format for speech string; 1 = way name; 2 = way route number"), name.addingXMLEscapes, code.addingXMLEscapes)
```

Comment:
This doesn’t work?

```swift
road = String.localizedStringWithFormat(NSLocalizedString("NAME_AND_REF", value: "%@ (%@)", comment: "Format for speech string; 1 = way name; 2 = way route number"), "<say-as interpret-as=\"address\">\(name.addingXMLEscapes)</say-as>", "<say-as interpret-as=\"address\">\(code.addingXMLEscapes)</say-as>")
```

(Split into multiple lines of course.)

It should have the same effect as the code you’ve written, except that now you’d be able to use the same localized format string for both cases and only vary the strings you end up inserting into the format string.